virt/xen/xen.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+
+import sys
+from time import time
+import XenAPI
+from XenAPI import Failure, NewMaster
+from collections import defaultdict, OrderedDict
+import virt
+
+class Guest(object):
```

Comment:
Sure, it's definitely a good start. If you could rebase it against current master and solve the issues I found, I think it will be almost ready to merge.